snap/snapcraft.yaml

```diff
@@ -43,18 +43,29 @@ hooks:
 
 environment:
   SNAP_LOG_DIR: ${SNAP_COMMON}/ops/snap/logs
+  OPENSEARCH_VARLIB: ${SNAP_COMMON}/var/lib/opensearch
+  OPENSEARCH_TMPDIR: ${SNAP_COMMON}/usr/lib/tmpfiles.d
```

Comment:
Probably `${SNAP_COMMON}/usr/share/tmp`? (this is [mainly for the `jna`](https://www.elastic.co/guide/en/elasticsearch/reference/current/executable-jna-tmpdir.html))